### DISCUSSION TITLE
Department Hours form interactivity possible bug

### DISCUSSION THREAD

**Discussion by 0ba100** [LiveHelperChat/livehelperchat#2066] [General]
Hi remdex, great software you have here. Excited to keep using it. I'm not paying for support presently, but I think I may have found a bug.

I've recently run the update script on the lhc instance I installed yesterday. I'm fuzzy on when it started happening, but on the department edit settings screen, selecting the days in the checklist is no longer opening up a form.

![lhc-bug](https://github.com/LiveHelperChat/livehelperchat/assets/75776440/54b6ccee-4806-4595-b79c-fb8570e5e25e)

I deployed your Digital Ocean setup (very convienient, thank you) and when I first installed it, it was working fine, only this morning it failed to work.

after it stopped working, I ran the update script and cleaned the cache, but saw no change in behavior.

Not sure how helpful this is, but it's seeming like when the change listener is called, `day` is null in this function:

```
function changeblock(inst) {
  let day = $(inst).attr("name");
  if ($(inst).is(":checked")) {
    $("." + day + "-block").removeClass("hide");
  } else {
    $("." + day + "-block").addClass("hide");
  }
}
```

and also, the input itself doesn't seem to have a "name" attribute. 

```
$('input[name="mod"]')
ce.fn.init {length: 0, prevObject: ce.fn.init}

$('input[data-name="mod"]')
ce.fn.init {length: 0, prevObject: ce.fn.init}
```

I can run `$('.mod-block').removeClass('hide');` and it works.

I looked at the code, and found this: 

https://github.com/LiveHelperChat/livehelperchat/commit/65c6050944f360ca5012f9fdda5223d6c6e0c2b4#r139640717

I hope this is helpful.

**Comment by 0ba100** [comment]
Happy to put in a PR if it's helpful, only it will take me a while to come up with something.

**Comment by 0ba100** [comment]
Looks like form submit is also broken, even when showing the form.

I show the forms, enter data, and click save, no custom time periods show up for my department.

**Reply by remdex** [reply]
You are running old version please upgrade properly. DigitalOcean has old lhc version. 

Upgrade sceipt there also can be old.
https://doc.livehelperchat.com/docs/upgrading
https://doc.livehelperchat.com/docs/install

**Reply by 0ba100** [reply]
I switched to using the docker setup on something I already host, and it's completely up to date per doc/CHANGELOG.md, but the behavior still presents.

**Reply by remdex** [reply]
I find hard to understand what exacrly you want to do. Can you record screen or try reproduce on demo.

**Reply by 0ba100** [reply]
https://github.com/LiveHelperChat/livehelperchat/assets/75776440/dc3ac1ed-1edd-46ea-8426-a7d4b0118059

Here is screen recording on demo

**Reply by remdex** [reply]
I see now. Will fix today :)

**Reply by remdex** [reply]
Fix I have done. New version with bunch of other stuff will be released late today.

**Comment by remdex** [comment]
And new version with a fix was released :)

**Comment by 0ba100** [comment]
Awesome, Thanks Remdex!
